include/llbuild/Core/BuildEngine.h

```diff
@@ -203,6 +203,16 @@ class BuildEngineDelegate {
   /// that the client can translate into an error.
   virtual Rule lookupRule(const KeyType& key) = 0;
 
+  /// Called when a cycle is detected by the build engine to check if it should
+  /// attempt to resolve the cycle and continue
+  ///
+  /// \param items The ordered list of items comprising the cycle, starting from
+  /// the node which was requested to build and ending with the first node in
+  /// the cycle (i.e., the node participating in the cycle will appear twice).
+  /// \returns True if the engine should attempt to resolve the cycle, false
+  /// otherwise.
```

Comment:
Probably good to have a little warning here about the consequences of returning true (or, alternately, we need that warning/discussion *somewhere*)